datastore/src/main/java/org/jboss/pnc/datastore/DefaultDatastore.java

```diff
@@ -419,10 +419,16 @@ private boolean isLatestSuccessBRFromThisBCA(BuildConfigurationAudited buildConf
      * Check is some of the dependencies from the previous build were rebuild.
      * Checking is done based on captured dependencies which are stored in the Build Record.
      */
-    private boolean hasARebuiltImplicitDependency(BuildRecord latestSuccessfulBuildRecord, boolean temporaryBuild) {
+    private boolean hasARebuiltImplicitDependency(BuildRecord latestSuccessfulBuildRecord, boolean temporaryBuild, IdRev idRev) {
```

Comment:
Adding a parameter that is used for logging only is not the best practice. If the parameter is not used in the logic, why is it important to log that. What about logging the operation where the method is called ?

Reply:
Yeah, I get you. I was trying to find out how to log in the cleanest way. I had a problem that this method didn't return any information about which implicit dependency triggered the rebuild.

Solution to this could be changing the signature of this method to return List<BuildRecord> and log it on a higher level.

Reply:
I see. Returning a list is also not nice. What about adding a log line (eg. checking if idRev hasARebuiltImplicitDependency) before the method call? Or leave it with the new parameter.

Reply:
I don't have a strong opinion on this. With the first option, there could be confusion in the situation when there is concurrent implicit checking done for 2 builds at the same time. You may run into a situation when you can't differentiate which implicit dependency belongs to which IdRev

Reply:
Something like this:
'''
line 1: Checking whether IdRev id:1 rev:1 has implicit dependencies that need rebuild.
line 2: Checking whether IdRev id:2 rev:2 has implicit dependencies that need rebuild.
line 3: Latest successful BuildRecord: 5 has has implicitly dependent BR: 4 that requires rebuild. (belongs to id:2 rev:2)
line 4 :Latest successful BuildRecord: 8 has has implicitly dependent BR: 3 that requires rebuild. (belongs to id:1 rev:1)
'''

Reply:
Parameter only for logging is the more justifiable the higher the logging level is. I personally would draw the line on the info level. Is this log message important enough to info level? If no I would be fine having it on two lines (thread id is usually part of the logline, so you can distinguish it).

Reply:
From my personal experience, this logline would save me 2-3 hours of finding out which implicitly-dependent build record caused the rebuild. I think it needs to be in DEBUG log level as we currently don't have any straightforward way of finding these build records.

Good catch, I've forgotten about thread id. With that, it should be perfectly fine having it in more lines. :+1: